src/main/java/com/springm/store/service/impl/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,95 @@
+package com.springm.store.service.impl;
+
+import com.springm.store.dto.cart.AddShoppingCartRequestDto;
+import com.springm.store.dto.cart.ShoppingCartDto;
+import com.springm.store.dto.cart.ShoppingCartResponseDto;
+import com.springm.store.exception.EntityNotFoundException;
+import com.springm.store.mapper.ShoppingCartMapper;
+import com.springm.store.model.Book;
+import com.springm.store.model.CartItem;
+import com.springm.store.model.ShoppingCart;
+import com.springm.store.model.User;
+import com.springm.store.repository.book.BookRepository;
+import com.springm.store.repository.cart.ShoppingCartRepository;
+import com.springm.store.repository.cart.item.CartRepository;
+import com.springm.store.repository.user.UserRepository;
+import com.springm.store.service.ShoppingCartService;
+import java.util.Optional;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    private final UserRepository userRepository;
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final CartRepository cartRepository;
+    private final BookRepository bookRepository;
+    private final ShoppingCartMapper shoppingCartMapper;
+    private final UserDetailsServiceImpl userDetailsService;
+
+    @Override
+    public ShoppingCartResponseDto getCart() {
+        Long userId = userDetailsService.getCurrentUserId();
+        User user = userRepository.findById(userId).orElseThrow(
+                () -> new IllegalArgumentException("User not found!"));
+        ShoppingCart cart = shoppingCartRepository.findByUser(user)
+                .orElseThrow(() -> new IllegalArgumentException("User not found!"));
```

Comment:
wrong message
add parameter to message
```suggestion
                .orElseThrow(() -> new EntityNotFoundException("User not found!"));
```